### DISCUSSION TITLE
How can I count the words on the page as the user of wikipedia sees it in the browser?

### DISCUSSION THREAD

**Discussion by dpriskorn** [earwig/mwparserfromhell#297] [Q&A] [answered]
I don't care about the references, but I want all the other templates expanded, can mwparserfromhell do that?
Or is there another way to get this count for a page?

**Comment by earwig** [comment, marked as answer]
Since you want to expand templates, this might be best done without mwparserfromhell. You can use the `action=render` endpoint to get the expanded HTML for a live page ([example](https://en.wikipedia.org/wiki/Earwig?action=render)) then [beautifulsoup](https://beautiful-soup-4.readthedocs.io/en/latest/) to extract the text:

```python
soup = bs4.BeautifulSoup(text)
count = len(' '.join(soup.stripped_strings)
```

**Reply by dpriskorn** [reply]
Thanks, i did not know about that parameter. 🙏
